paasta_tools/kubernetes_tools.py

```diff
@@ -898,14 +877,47 @@ def get_autoscaling_metric_spec(
                         ),
                     ),
                 )
-            )
-        else:
+            ]
+
+        log.error(
+            f"Unknown metrics_provider specified: {provider['type']} for\
+            {name}/name in namespace{namespace}"
+        )
+        return []
+
+    def get_autoscaling_metric_spec(
+        self,
+        name: str,
+        cluster: str,
+        kube_client: KubeClient,
+        namespace: str,
+    ) -> Optional[V2beta2HorizontalPodAutoscaler]:
+        # Returns None if an HPA should not be attached based on the config,
+        # or the config is invalid.
+
+        if self.get_desired_state() == "stop":
+            return None
+
+        if not self.is_autoscaling_enabled():
+            return None
+
+        autoscaling_params = self.get_autoscaling_params()
+        if autoscaling_params["metrics_providers"][0]["decision_policy"] == "bespoke":
+            return None
```

Comment:
should we perhaps add some validation so that folks can't mix bespoke autoscaling with HPA-based autoscaling? i can see someone accidentally bypassing this check by adding an HPA-based provider above an existing bespoke provider

Reply:
Yea, good callout.  I'll add that to my list of things to validate.

Reply:
Done